packages/site/cms-site/src/blocks/factories/BlocksBlock.tsx

```diff
@@ -31,8 +35,11 @@ export const BlocksBlock: React.FC<Props> = ({ supportedBlocks, data: { blocks }
 
                     return null;
                 }
-
-                return <React.Fragment key={block.key}>{blockFunction(block.props)}</React.Fragment>;
+                return (
+                    <React.Fragment key={block.key}>
+                        <ErrorHandlerBoundary onError={onError}>{blockFunction(block.props)}</ErrorHandlerBoundary>
```

Comment:
the ErrorHandlerBoundary could get the onError itself, that would make using this component simpler.

(maybe you need an internal wrapper that uses useErrorHandler as it is a class component)

Reply:
> maybe you need an internal wrapper that uses useErrorHandler as it is a class component

Oh that is smart I didn't think of that.

I currently tried it here like that:
201e6a8eeff2e3504244270cef33ac1e5c6d068b

and changed it a bit in this commit to also have the errorInfo:
0297499986541e53708e3ebfd1eb6f7eaf8d5e21

Did you mean it like that @nsams, or did you mean a wrapper in the componentDidCatch ? Because I couldn't find a way to do it directly in there. 

It feels a bit hacky at the moment, because the errorInfo is only available in the `componentDidCatch` but not in `getDerivedStateFromError` . And according to the docs [here](https://react.dev/reference/react/Component#catching-rendering-errors-with-an-error-boundary), `getDerivedStateFromError` should be used to update the state and `componentDidCatch` for reporting the error.

And currently I also have the problem that the error is now reported multiple times.

Any suggestions? @johnnyomair @nsams

Reply:
No, doing that during render seems wrong (and can easily cause the double calls you describe)

I was thinking about something like that:
```
//old class, now internal
export class ErrorHandlerBoundaryInternal ....

//new wrapper function component, that is the exported
export const ErrorHandlerBoundary = (.....) => {
    const { onError } = useErrorHandler();
    return <ErrorHandlerBoundaryInternal onError={onError}>{children}</ErrorHandlerBoundaryInternal>;
}
```

Reply:
Thank you, I could have thought of that too...

done in:
065f3cbaca7ffacedd28c1429a9c1da2ea0266ac